python/pyarrow/error.pyx

```diff
@@ -19,13 +19,52 @@ from pyarrow.includes.libarrow cimport CStatus
 from pyarrow.includes.common cimport c_string
 from pyarrow.compat import frombytes
 
+
 class ArrowException(Exception):
     pass
 
+
+class ArrowInvalid(ValueError, ArrowException):
```

Comment:
Isn't this an antipattern (deriving from a built-in exception or, worse, raising them directly)? If a user combines some Arrow code with their own in a try block, they may inadvertently catch, say, an `ArrowInvalid` Exception when they explicitly were trying to handle a `ValueError` for non-arrow code in the try block.

tl;dr if you structure your Exceptions like this, client code has no idea from where the issue originated and has no easy way of disambiguating in the `except` block (if they're even aware of the Arrow Exception class hierarchy to begin with).